db/view/view.cc

```diff
@@ -478,41 +480,187 @@ row_marker view_updates::compute_row_marker(const clustering_row& base_row) cons
         return cell.is_live_and_has_ttl() ? row_marker(cell.timestamp(), cell.ttl(), cell.expiry()) : row_marker(cell.timestamp());
     }
 
-    return marker;
+    return base_row.marker();
 }
 
-deletable_row& view_updates::get_view_row(const partition_key& base_key, const clustering_row& update) {
-    std::vector<bytes> linearized_values;
-    auto get_value = boost::adaptors::transformed([&, this] (const column_definition& cdef) -> managed_bytes_view {
-        auto* base_col = _base->get_column_definition(cdef.name());
+
+namespace {
+struct bytes_view_with_action {
+    managed_bytes_view _bytes_view;
+    bytes_with_action::action _action;
+    bytes_view_with_action(managed_bytes_view view, bytes_with_action::action action)
+        : _bytes_view(view)
+        , _action(action)
+    {}
+    bytes_view_with_action(managed_bytes_view view)
+        : _bytes_view(view)
+    {}
+    bytes_view_with_action(bytes_with_action&& bwa, std::deque<bytes>& linearized_values)
+        : bytes_view_with_action(managed_bytes_view(linearized_values.emplace_back(std::move(bwa._view))), bwa._action)
+    {}
+    static managed_bytes_view get_view(const bytes_view_with_action& bvwa) {
+        return bvwa._bytes_view;
+    }
+};
+
+// value_getter is used to extract values for specific columns during view update.
+struct value_getter {
+    // linearized_values hold bytes for values of computed columns, for which we later store references in the form of managed_bytes_view.
+    // deque doesn't invalidate references at emplace_back.
+    std::deque<bytes> linearized_values;
```

Comment:
What happens if a collection value causes a clustering key that is too large to be generated? Or is that impossible?

Reply:
This is a good question. If I remember correctly, while old Cassandra had a 64KB limit on collection values, this limit was lifted, while clustering keys do have a 64KB limit.
I'll write a test for this, and fix it if it doesn't have reasonable behavior.

Reply:
https://github.com/scylladb/scylla/issues/10948

Reply:
That was a duplicate. The original issue: https://github.com/scylladb/scylla/issues/8627
I'm working now on the tests, and the fix.

Reply:
To clarify, the current code - in both regular indexing and the new collection indexing - can indeed generate an oversized clustering key. The write of this clustering key will fail. But the way it fails looks like an obscure internal server error instead of a clean InvalidRequest saying that the query is broken and why.
It's not a disaster (Scylla doesn't crash...) but it still needs to be fixed, and I'm planning to add another patch to fix it (both in the collection indexing case and in the regular indexing case).

Reply:
@eliransin @psarna FYI I'm working on fixing this comment from Avi (which is https://github.com/scylladb/scylla/issues/8627) right now. It's taking me longer than I hoped because I'm rusty with the materialized views code, which has also become so complex that it's really hard to get anything done there :-(

Reply:
I decided that fixing https://github.com/scylladb/scylla/issues/8627 is harder than it's worth, so will not do it now. Instead I will add to this series as a 28th patch more test cases for this issue (oversized items in a collection that can't be indexed). The test will xfail, but will demonstrate that the oversized keys still fail in an "acceptable" manner (replica write error) and not cause any crashes or other disasters.
@avikivity do you think that's acceptable?

Reply:
yes